tests/test__mongomock.py

```diff
@@ -1743,6 +1743,46 @@ def test__aggregate26(self):
         ]
         self.cmp.compare_ignore_order.aggregate(pipeline)
 
+    def test__aggregate27(self):
+        # group addToSet
+        pipeline = [
+            {"$group": {"_id": "$a", "nb": {"$addToSet": "$count"}}},
+            {"$sort": {"_id": 1}}
+        ]
+        # self.cmp.compare cannot be used as addToSet returns elements in an unpredictable order
+        aggregations = self.cmp.do.aggregate(pipeline)
+        expected = list(aggregations["real"])
+        result = list(aggregations["fake"])
+        assert len(result) == len(expected)
+        for i, elt in enumerate(result):
+            for k, v in elt.iteritems():
+                if type(v) is list:
+                    assert set(v) == set(result[i][k])
+                else:
+                    assert v == result[i][k]
+
+    def test__aggregate28(self):
+        # group addToSet dict element
+        self.cmp.do.remove()
+        data = [
+            {"a": {"c": "1", "d": 1}, "b": {"c": "2", "d": 2}, "nb": 1},
+            {"a": {"c": "3", "d": 3}, "b": {"c": "4", "d": 4}, "nb": 1},
+            {"a": {"c": "5", "d": 1}, "b": {"c": "6", "d": 6}, "nb": 2}
```

Comment:
There's not much of a case here: make sure to add at least two items that have the same "a.c", a different "a.d" but the same "b" entirely.